### DISCUSSION TITLE
Submitting Blobs using Wagmi

### DISCUSSION THREAD

**Discussion by remarcable** [wevm/wagmi#3754] [Q&A]
Hi there 👋 

It is possible with `viem` to [submit Blob Transactions](https://viem.sh/docs/guides/blob-transactions).

As `wagmi` is using `viem` under the hood, I would have expected to be able to seamlessly do the same.

So instead of [using `viem`](https://viem.sh/docs/guides/blob-transactions#send-blob-transaction) like this:

```ts
const blobs = toBlobs({ data: stringToHex('hello world') })
 
const hash = await client.sendTransaction({
  blobs,
  kzg,
  maxFeePerBlobGas: parseGwei('30'),
  to: '0x0000000000000000000000000000000000000000',
})
```

I expected to be able to do the same with `wagmi` like this:

```ts
  const { sendTransactionAsync } = useSendTransaction();

  const sendBlobTransaction = useCallback(async () => {
    const kzg = await getKZG();

    return sendTransactionAsync({
      blobs: toBlobs({ data: stringToHex('Some random blob data') }),
      kzg,
      maxFeePerBlobGas: parseGwei('200'),
      to: '0x0000000000000000000000000000000000000000',
    });
  }, [sendTransaction]);
```

When this runs, instead of an `EIP-4844` transaction, it submits an `EIP-1559` transaction.

It's not really important if I use `useSendTransaction` or `sendTransaction` from `@wagmi/core`. Only `viem`'s `sendTransaction` does this correctly. The reason I want to use `wagmi` instead of `viem` is the ease of connecting different wallets and of course the awesome React hooks.

One solution I'm looking into right now is using `viem` directly through a [JSON-RPC account](https://viem.sh/docs/accounts/jsonRpc). But using `viem` directly feels somewhat against the purpose of `wagmi`.

I've created a reproduction here: https://stackblitz.com/edit/stackblitz-starters-yajwnr?file=app%2Fpage.tsx

Any idea why it doesn't pass `kzg` and `blobs` down to `viem`? Are they filtered out somewhere?

Thanks! I appreciate your work, your time, and your help 🙏  Please let me know if I should resubmit this as a GitHub issue.

**Comment by jeasonstudio** [comment]
+1 need this.
